packages/optimistic-oracle/src/proposer.js

```diff
@@ -286,159 +266,126 @@ class OptimisticOracleProposer {
         this.utf8ToHex(priceRequest.identifier),
         priceRequest.timestamp,
         priceRequest.ancillaryData
       );
-
-      // Simple version of inventory management: simulate the transaction and assume that if it fails,
-      // the caller didn't have enough collateral.
-      let disputeBond, gasEstimation;
-      try {
-        [disputeBond, gasEstimation] = await Promise.all([
-          dispute.call({ from: this.account }),
-          dispute.estimateGas({ from: this.account })
-        ]);
-      } catch (error) {
-        this.logger.error({
-          at: "OptimisticOracleProposer#sendDisputes",
-          message: "Cannot dispute price: not enough collateral (or large enough approval)✋",
-          proposer: this.account,
-          disputeBond,
-          priceRequest,
-          error
-        });
-        return;
-      }
-      const txnConfig = {
-        from: this.account,
-        gas: Math.min(Math.floor(gasEstimation * this.GAS_LIMIT_BUFFER), this.txnGasLimit),
-        gasPrice: this.gasEstimator.getCurrentFastPrice()
+      const transactionConfig = {
+        gasPrice: this.gasEstimator.getCurrentFastPrice(),
+        from: this.account
       };
-
       this.logger.debug({
         at: "OptimisticOracleProposer#sendDisputes",
         message: "Disputing proposal",
         priceRequest,
         proposalPrice,
-        disputeBond,
         disputePrice,
         allowedError: this.disputePriceErrorPercent,
-        txnConfig
+        transactionConfig
       });
-
-      // Send the transaction or report failure.
-      let receipt;
       try {
-        receipt = await dispute.send(txnConfig);
+        const transactionResult = await runTransaction({
+          transaction: dispute,
+          config: transactionConfig
+        });
+        let receipt = transactionResult.receipt;
+        let returnValue = transactionResult.returnValue;
+
+        const logResult = {
+          tx: receipt.transactionHash,
+          requester: receipt.events.DisputePrice.returnValues.requester,
+          proposer: receipt.events.DisputePrice.returnValues.proposer,
+          disputer: receipt.events.DisputePrice.returnValues.disputer,
+          identifier: this.hexToUtf8(receipt.events.DisputePrice.returnValues.identifier),
+          ancillaryData: receipt.events.DisputePrice.returnValues.ancillaryData,
+          timestamp: receipt.events.DisputePrice.returnValues.timestamp,
+          proposedPrice: receipt.events.DisputePrice.returnValues.proposedPrice
+        };
+        this.logger.info({
+          at: "OptimisticOracleProposer#sendDisputes",
+          message: "Disputed proposal!⛑",
+          priceRequest,
+          disputePrice,
+          disputeBond: returnValue,
+          allowedError: this.disputePriceErrorPercent,
+          transactionConfig,
+          disputeResult: logResult
+        });
       } catch (error) {
+        const message =
+          error.type === "call"
```

Comment:
how hard it is to make a test case to check this is working, ie check both branches of the log message, so it would have caught the issue of this not setting correctly?

Reply:
I'd like to unit test the `common/ContractUtils/runTransaction` method but I'm having trouble thinking of a situation where `.call()` succeeds but `.send()` fails... any ideas? @chrismaree @mrice32

Reply:
I added a unit test for `runTransaction` and left a TODO for how to test the `.send()` path

Reply:
i see yea, that makes sense. its not a huge deal though